src/execution/execution_entry_point.rs

```diff
@@ -578,4 +628,170 @@ impl ExecutionEntryPoint {
+            caller_address: self.caller_address.clone(),
+            entry_point_selector: self.entry_point_selector.clone(),
+            tx_execution_context: tx_execution_context.clone(),
+            block_context: block_context.clone(),
+        };
+
+        native_program
+            .insert_metadata(SyscallHandlerMeta::new(&syscall_handler))
+            .unwrap();
+
+        let syscall_addr = native_program
+            .get_metadata::<SyscallHandlerMeta>()
+            .unwrap()
+            .as_ptr()
+            .addr();
+
+        let fn_id = &sierra_program
+            .funcs
+            .iter()
+            .find(|x| x.id.id == (entry_point.function_idx as u64))
+            .unwrap()
+            .id;
+
+        let number_of_params = sierra_program.funcs[fn_id.id as usize].params.len();
+
+        let required_init_gas = native_program.get_required_init_gas(fn_id);
+
+        let calldata: Vec<_> = self
+            .calldata
+            .iter()
+            .map(|felt| felt252_bigint(felt.to_bigint()))
+            .collect();
+
+        /*
+            Below we construct `params`, the Serde value that MLIR expects. It consists of the following:
+
+            - One `null` value for each builtin that is going to be used.
+            - The maximum amout of gas allowed by the call.
+            - `syscall_addr`, the address of the syscall handler.
+            - `calldata`, an array of Felt arguments to the method being called.
+
+            The code to construct it is complex and obscure for the following reason: because the amount of syscalls
+            to be called depends on the method, we don't know beforehand how many `null`s to append to the beginning
+            of the parameters. Ideally, we would like to use the `json!` macro to construct the `params`, but I found
+            no way of doing so; thus, string interpolation.
+
+            To know how many `null`s to append, we fetch the total amount of arguments for the function from
+            the sierra program, then substract 3 from it, to account for the other arguments (gas, syscall_addr, calldata).
+        */
+        let mut builtins_string = "".to_owned();
+        for _ in 0..(number_of_params - 3) {
+            builtins_string.push_str("null,");
+        }
+
+        let params: Value = serde_json::from_str(&format!(
+            "
+            [
+                {}
+                {},
+                {},
+                [
+                    {:?}
+                ]
+            ]
+        ",
+            builtins_string,
+            u64::MAX,
+            syscall_addr,
+            calldata
+        ))
+        .unwrap();
+
+        let mut writer: Vec<u8> = Vec::new();
+        let returns = &mut serde_json::Serializer::new(&mut writer);
+
+        let native_executor = NativeExecutor::new(native_program);
+
+        native_executor
+            .execute(fn_id, params, returns, required_init_gas)
+            .unwrap();
```

Comment:
This could crash the server.

Reply:
I mapped the error into a custom error. I hope that's fine.